subsys/bluetooth/mesh/pb_gatt_srv.c

```diff
@@ -244,9 +244,12 @@ static size_t gatt_prov_adv_create(struct bt_data prov_sd[2])
 
 dev_name:
 	if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT_USE_DEVICE_NAME)) {
+		uint8_t name[BT_GAP_DEVICE_NAME_MAX_SIZE];
```

Comment:
I'm pretty sure after {} `name` array will disappear and stack might be used for anything else. Device might have garbage instead of name.

Even more, if you allocate memory on stack within this function, after leaving it the stack will be released and name will be again garbage.

This memory should be kept until Host copies in its buffer or global.
Probably place where `struct bt_data` is allocated should be sufficient.

Reply:
You are right, I did the update too quickly. It should be good now.